src/plugins/intel_cpu/src/nodes/mkldnn_matmul_node.cpp

```diff
@@ -22,9 +22,357 @@
+                uni_vfmadd231ps(get_dst_vmm(amount_full), get_src_vmm(0), get_src_vmm(1));
+                add(reg_src_aux_1, amount_tail * sizeof(float));
+            }
+
+            add(reg_src_aux_0, sizeof(float));
+            add(k, 1);
+            jmp(label_k, T_NEAR);
+        }
+        L(label_k_end);
+
+        for (int i = 0; i < amount_full; ++i) {
+            apply_post_ops(get_dst_vmm(i).getIdx());
+            store_ptr(ptr[reg_dst + i * vlen], get_dst_vmm(i));
+        }
+        add(reg_dst, amount_full * vlen);
+
+        if (amount_tail != 0) {
+            apply_post_ops(get_dst_vmm(amount_full).getIdx());
+            store(get_dst_vmm(amount_full), reg_dst, amount_tail);
+            add(reg_dst, amount_tail * sizeof(float));
+        }
+    }
+
+    // optimized execution for cases with transposed matrix b or k = 1
+    inline void optimized_body_loop() {
+        Xbyak::Label label_n;
+        Xbyak::Label label_n_end;
+
+        const int full_n_amount = jcp_.n / vec_step;
+        const int tail_n_amount = jcp_.n % vec_step;
+
+        mov(reg_dst_aux, reg_dst);
+        mov(reg_src_aux_1, reg_src_1);
+
+        mov(n, 0);
+        L(label_n); {
+            cmp(n, full_n_amount);
+            je(label_n_end, T_NEAR);
+
+            optimized_vectorization_body_loop(vec_step);
+
+            if (attr_.post_ops_.len() != 0) {
+                load(reg_dst_aux, get_dst_vmm(0), vec_step);
+                apply_post_ops();
+                store(get_dst_vmm(0), reg_dst_aux, vec_step);
+                add(reg_dst_aux, vlen);
+            }
+
+            add(n, 1);
+            jmp(label_n, T_NEAR);
+        }
+        L(label_n_end);
+
+        if (tail_n_amount != 0) {
+            optimized_vectorization_body_loop(tail_n_amount);
+
+            if (attr_.post_ops_.len() != 0) {
+                load(reg_dst_aux, get_dst_vmm(0), tail_n_amount);
+                apply_post_ops();
+                store(get_dst_vmm(0), reg_dst_aux, tail_n_amount);
+                add(reg_dst_aux, tail_n_amount * sizeof(float));
+            }
+        }
+    }
+
+    inline void optimized_vectorization_body_loop(const int amount) {
+        Xbyak::Label label_o;
+        Xbyak::Label label_o_end;
+
+        mov(n_inside, 0);
+        L(label_o); {
+            cmp(n_inside, amount);
+            je(label_o_end, T_NEAR);
+
+            mov(reg_src_aux_0, reg_src_0);
+            uni_vpxor(get_dst_vmm(), get_dst_vmm(), get_dst_vmm());
+
+            for (int i = 0; i < amount_full; ++i) {
+                load_ptr(get_src_vmm(0), ptr[reg_src_aux_0 + i * vlen]);  // src0
+                load_ptr(get_src_vmm(1), ptr[reg_src_aux_1 + i * vlen]);  // src1
+
+                uni_vfmadd231ps(get_dst_vmm(), get_src_vmm(0), get_src_vmm(1));
+            }
+            add(reg_src_aux_1, amount_full * vlen);
+
+            if (amount_tail != 0) {
+                add(reg_src_aux_0, amount_full * vlen);
```

Comment:
Why `reg_src_aux_0` is incremented inside the condition while `reg_src_aux_1` outside?